specification/communication/data-plane/Rooms/preview/2022-02-01-preview/communicationservicesrooms.json

```diff
@@ -0,0 +1,707 @@
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "x-ms-client-name": "ErrorCode",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete a meeting room.": {
+            "$ref": "./examples/Rooms_DeleteRoom.json"
+          }
+        }
+      }
+    },
+    "/rooms/{roomId}/participants": {
+      "get": {
+        "tags": [
+          "Participants"
+        ],
+        "summary": "Get participants in a room.",
+        "operationId": "Rooms_GetParticipants",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "roomId",
+            "description": "The id of the room to get participants from",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The action returns the participants of a room.",
+            "schema": {
+              "$ref": "#/definitions/ParticipantsCollection"
+            }
+          },
+          "default": {
+            "description": "Error response",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "x-ms-client-name": "ErrorCode",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get participants": {
+            "$ref": "./examples/Rooms_GetParticipants.json"
+          }
+        }
+      }
+    },
+    "/rooms/{roomId}/participants:add": {
+      "post": {
+        "tags": [
+          "Participants"
+        ],
+        "summary": "Adds participants to a room. If participants already exist, no change occurs.",
+        "operationId": "Rooms_AddParticipants",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "roomId",
+            "description": "Id of the room to add participants to.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "addParticipantsRequest",
+            "description": "Participants to be added to the room.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AddParticipantsRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The participants were successfully added.",
```

Comment:
nit: Document the returned `RoomModel` as in other APIs.

Also, should these `participants:<verb>` APIs return a `RoomModel`, or just the list of participants? It's a bit weird that the response for adding a participant includes information like room validity.

Reply:
nit addressed. 
Retaining current behavior to return RoomModel for any update to a room, including update to room participants to provide a complete view to the customer how their room looks like after the action across the rooms API.

Reply:
I'd recommend against this confusion between models in your API.

As a precedent, see the chat service's participants:add - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/Chat/stable/2021-09-07/communicationserviceschat.json#L1527

It returns an object that contains information about the add operation itself, but no information about even all the participants.

This is inline with CRUD operations: Create operation returns the created model, Read of course returns it as well, but Update and Delete do not.

Here's one reason why returning the RoomModel here might be bad: Let's say you want to implement `filter` for GET on `room/<id>`. Now the `participants:add` becomes another method that also returns the same RoomModel, but does not implement `filter`. This will cause confusion in the API surface, and might actually be bad if the `filter` is needed for performance reasons.

Another performance argument: Contoso decides to add 200 participants one by one with `participants:add`. Then we end up returning the room back each time, with increasing number of participants. Effectively, our cumuilative response size will be O(N^2) in the number of participants.

Reply:
@prprabhu-ms Addressed. The response end points of Participants action end points has been updated to return participants collection instead of a room.